src/app/Http/Controllers/TweetController.php

```diff
@@ -48,36 +49,33 @@ public function show(int $userId)
     }
 
     /**
-     * ツイート投稿
+     * 認証ユーザーのみツイート投稿
      *
-     * @param  \Illuminate\Http\Request  $request
-     * @return \Illuminate\Http\Response
+     * @param  PostRequest $request
+     * @param  Tweet $tweet
+     * @param  User $user
      */
-    public function store(Request $request)
+    public function store(PostRequest $request, Tweet $tweet, User $user)
     {
-        //
-    }
+        $isAuthUser = Gate::forUser($request->user())->allows('store-tweet', $tweet);
+        $user->checkIsAuth($isAuthUser);
```

Comment:
しっかり認可の部分を考えられてて素晴らしいね！！！

実装面でアドバイスすると、UserクラスでcheckIsAuthメソッドを定義していると思うんだけど、
ModelでResponseを返す実装をするのは責務上あまり好ましくないので、
checkIsAuthメソッドではbool値だけ返して、Responseの実装はControllerレイヤーで実装したほうがいいね！
```suggestion
       if (!$user->checkIsAuth($isAuthUser)) {
           abort(Response()->json(['error' => '認証されていないユーザーです。'], 401));
       }
```

Reply:
bool返す部分のみModelで書いて、ResponseはControllerで実装しました！